---
title: "eDNA Spiny Lobster All R analysis"
author: "Aiko Abo Dominguez"
date: "2025-06-09"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

installing and loading packages
```{r}
install.packages(c("ggplot2",
                   "ape",
                   "vegan",
                   "tidyverse"))
install.packages("sf")
install.packages("mapview")

install.packages('devtools')
library(devtools)

install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")
devtools::install_github("jbisanz/qiime2R")

#installing limma
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install(version='devel')
BiocManager::install("limma")
BiocManager::install("phyloseq")
BiocManager::install("microbiome")

library(qiime2R)
library(sf)
library(mapview)
library(tidyverse)
library("phyloseq")
library("ggplot2")
library("ape")
library("vegan")
library(tidyr)
library(dplyr)
library(tibble)
library("pairwiseAdonis"); packageVersion("pairwiseAdonis")
library(limma)
library(microbiome)  
```

Importing and editing metadata file

```{r}
#import file
map <- read.csv("/Users/aiko.abo.dominguez/Desktop/Spiny_lobster_project/metadata_lobster_count24.csv")

#change the name of the first column
map$sample_id_old <- map$sample.id
colnames(map)[1] <- "sample_id"
colnames(map)

#make function to truncate ID name
truncate_before_second_underscore <- function(x) {
  sapply(strsplit(x, "_"), function(parts) paste(parts[1:2], collapse = "_"))
}

map$sample_id <- truncate_before_second_underscore(map$sample_id)
View(map)

#set column 1 as row names
rownames(map) <- map$sample_id

#make sure it is a dataframe
map <- as.data.frame(map)
View(map)

#setting variables to be numeric, as appropriate
map$Lat <- as.numeric(map$Lat)
map$Long <- as.numeric(map$Long)
map$Total_Lobsters <- as.numeric(map$Total_Lobsters)
map$EstimatedBiomass <- as.numeric(map$EstimatedBiomass)

str(map)

#add column for sample site ID based on latitude
sites <- unique(map$Lat)
sites
map <- map %>%
  mutate(site_id = ifelse(Lat==sites[1], "A",
                   ifelse(Lat==sites[2], "B",
                   ifelse(Lat==sites[3], "C",
                   ifelse(Lat==sites[4], "D",
                   ifelse(Lat==sites[5], "E",
                   ifelse(Lat==sites[6], "F", 
                   ifelse(Lat==sites[7],"G",
                   ifelse(Lat==sites[8], "H",
                   ifelse(Lat==sites[9], "I",
                   ifelse(Lat==sites[10], "J", 
                   ifelse(Lat==sites[11], "K",
                   ifelse(Lat==sites[12], "L", NA)))))))))))))
unique(map$site_id)
unique(map$Don)
View(map)
```

Examining metadata
```{r}
#plotting sample long and lat visually
mapview(map, xcol = "Long", ycol = "Lat", crs = 4269, grid = FALSE)

```


Importing ASV sequences, taxa table, and phylogenetic tree
```{r}
#ASV table from dada2 of forward reads, only eukaryotes
ASV_qza <- read_qza("/Users/aiko.abo.dominguez/Desktop/Spiny_lobster_project/table_ASVs_lobC16S_euks_fwd.qza")
ASV_table <- as.matrix(ASV_qza$data)
View(ASV_table)

#truncating ASV column names to match "map" truncated sample names
colnames(ASV_table) <- truncate_before_second_underscore(colnames(ASV_table))
colnames(ASV_table)
map[,"sample_id"]

#check that it worked
all(rownames(map)%in%colnames(ASV_table))

#import taxa from dada2 of forward reads
taxa_qza <- read_qza("/Users/aiko.abo.dominguez/Desktop/Spiny_lobster_project/taxonomy_fwd.qza")
taxa_table <- taxa_qza$data 
taxa_table <- taxa_table %>%
  as_tibble() %>%
  separate(Taxon, sep=";", c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")) %>%
  column_to_rownames("Feature.ID") %>%
  as.matrix()
  
View(taxa_table)

#Importing phylogenetic tree
tree_file_qza <-read_qza("/Users/aiko.abo.dominguez/Desktop/Spiny_lobster_project/rooted_masked_aligned_seq_ASVs_lobC16S_euks_fwd.qza")
tree_file<- tree_file_qza$data
View(tree_file)

```

Generating phyloseq
```{r}
#generating object
ps <- phyloseq(otu_table(t(ASV_table), taxa_are_rows=FALSE),
              sample_data(map), 
              tax_table(taxa_table), 
              tree_file)
ps
```

Tried to filter taxa length by site but it didn't work because i can't recombine them into one ps since now they have different branches
```{r}


#filter by site
ps_by_site <- list()

for (i in 1:length(sites)) {
  site <-map$site_id[i]
  ps_by_site[[site]] <- subset_samples(ps, site_id==site)
}

#filter taxa in each site
ps_by_site_filtered <- list()

for (i in 1:length(sites)) {
  site <-map$site_id[i]
  ps_by_site_filtered[[site]] <- filter_taxa(ps_by_site[[site]], function(x) sum(x > 10) > (0.30*length(x)), TRUE)
}

View(ps_by_site_filtered)

#combine back to one ps DOESNT WORK
#ps_filtered <- do.call("merge_phyloseq", ps_by_site_filtered)
```

Filter all ps taxa
```{r}
#filtering taxa. Keeps ASVs where its abundance is greater than 10 in more than 10% of samples. Adjust to see what happens. 30% looks like it cuts out a lot (8879 taxa to 540), so maybe lower this?
ps_filter <- filter_taxa(ps , function(x) sum(x > 10) > (0.10*length(x)), TRUE)
ps_filter
```

CLR transformation for filtered ps

```{r}
ps_clr <- microbiome::transform(ps_filter, 'clr')
ps_clr
```

Relative abundance computed using custom function, sample-wise (compositional data) by ORDER for filtered  ps I think this one and the previous one do the same thing
```{r}
#convert ps absolute ASV count into relative abundances by classification level
ps_ra = transform_sample_counts(ps_filter, function(x) x / sum(x)) %>%
tax_glom("Order")%>% psmelt()

#View(ps_ra)
dim(ps_ra)
```

Identifying for NAs in ps filtered by taxonomic rank and turned into a df
```{r}
order_NA <- subset(ps_ra, Order=="NA")
family_NA <- subset(ps_ra, Family=="NA")
genus_NA <- subset(ps_ra, Genus=="NA")
dim(order_NA)
dim(family_NA)
dim(genus_NA)

#View(family_NA)

#OTUs that are NA in order, but are not in genus
setdiff(order_NA$OTU,genus_NA$OTU)

#OTUs that are NA genus, but not order
setdiff(genus_NA$OTU,order_NA$OTU)

#this OTU is not in Order, but is in Genus
sum(genus_NA$OTU=="a9d4ae9f0286fe9bdb3881218ea6b4c1")
```

Removing NAs in ps filtered by taxonomic rank and turned into a df
```{r}
#adding back Order to missing ones, had to look up by hand
ps_ra$Order[ps_ra$OTU == "a9d4ae9f0286fe9bdb3881218ea6b4c1"] <- "Neotaenioglossa"
ps_ra$Order[ps_ra$OTU == "8ab5f93e9ac490a760599fb46ab9f46a"] <-'Hubrechtiiformes'

#check that it worked
order_NA <- subset(ps_ra, Order=="NA")
View(order_NA)

#removing rest of NAs? OR REPLACE WITH "UNIDENTIFIED"
#ps_ra <- ps_ra[ps_ra$Order != "NA", ]

```

Set colors for graph
```{r}
#choose which ps to visualize
plot_ps <- ps_ra


#loading colors

library(RColorBrewer)
nb.cols <- length(unique(plot_ps$Order)) #set taxonomic rank
nb.cols


#set color palette, maybe better suited to more  taxonomic ranks?
mycolors <- colorRampPalette(c(brewer.pal(9, "Set1"),
                               brewer.pal(8, "Set2"),
                               brewer.pal(12, "Set3")))(nb.cols)

```

Making relative abundance graph of orders present
```{r}
RA_sup = plot_ps %>%
subset(Abundance >0.001) %>% #filter for abundance
#mutate_if(
#                is.character, 
#                stringr::str_replace_all, 
#                pattern = "o__",
#                replacement = "")  %>%

ggplot(  
       aes(x = Sample, y=Abundance, 
           fill=Order)) + #fill is  taxonomic rank

geom_bar(stat="identity", position="fill") +
scale_fill_manual(values = mycolors) +
guides(fill = guide_legend(keywidth = 0.5, , keyheight =.60, ncol=1)) +
theme_classic() + #theme
facet_grid(.~Lat, scales="free") + #separate by lattidue ("Lat")

#text sizes
theme(legend.text =element_text(size=4)) + 
theme(legend.title =element_text(size=4)) +
theme(axis.title.x = element_text(size = 5)) +
theme(axis.text.x = element_text(size =4)) +
theme(axis.title.y = element_text(size = 5)) +
theme(axis.text.y = element_text(size =4)) +
ylab("Relative Abundance") +
xlab("Sample")

RA_sup
#ggsave('spiny_lobster_edna_order.png', plot=RA_sup,
#       width = 14,
#       height = 7,
#       path = "/Users/aiko.abo.dominguez/Desktop/Spiny_lobster_project/R_code/figures")
```
Compute distance matrix based on the OTU/ASV table using Euclidean distance to assess beta diversity
```{r}
dis <- vegdist(otu_table(ps_clr), method ="euclidean")
#test homogeneity of group dispersions (variances).
mod <- betadisper(dis, sample_data(ps_clr)$site_id)
mod
## Permutation test for F
permutest(mod)
```
not sure if this works- come back to it
```{r}
#convert meta data to dataframe
df_clr= as(sample_data(ps_clr), "data.frame")

#All samples
adonis(otu_table(ps_clr)~ site_id, 
       data =df_clr, permutations = 999, 
         method = "euclidean")
```

Perform RDA and plot it
```{r}
ps.ord_clr <- ordinate(ps_clr, "RDA", "euclidean")
```

Choose RDA plot colors
```{r}
mycolors <- colorRampPalette(c("red", "yellow"))
```


Plot RDA
```{r}
beta = plot_ordination(ps_clr, ps.ord_clr,color="site_id") +
       scale_color_manual(values=mycolors(12)) +
       theme_classic() +
       theme(legend.text =element_text(size=4)) +
       theme(legend.title = element_text(size=4)) +
       theme(axis.title.x = element_text(size = 5)) +
       theme(axis.title.y = element_text(size = 5)) +
       theme(axis.text.x = element_text(size =4)) +
       theme(axis.text.y = element_text(size =4)) +
       labs(color='Site ID') 
# theme(legend.position = "none",
 #     panel.grid.minor.x = element_blank())

beta
```

