---
title: "eDNA Spiny Lobster All R analysis"
author: "Aiko Abo Dominguez"
date: "2025-06-09"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Downloads
installing and loading packages
```{r installing and loading packages}
# Function to install CRAN packages if not already installed
install_if_missing <- function(pkgs) {
  installed <- installed.packages()[, "Package"]
  to_install <- setdiff(pkgs, installed)
  if (length(to_install) > 0) {
    install.packages(to_install)
  }
}

# CRAN packages
install_if_missing(c("ggplot2", "ape", "vegan", "tidyverse", "sf", "mapview"))

# devtools (for GitHub installs)
install_if_missing("devtools")
library(devtools)

# GitHub packages
if (!"pairwiseAdonis" %in% installed.packages()[, "Package"]) {
  install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")
}

if (!"qiime2R" %in% installed.packages()[, "Package"]) {
  install_github("jbisanz/qiime2R")
}

# Bioconductor setup
if (!"BiocManager" %in% installed.packages()[, "Package"]) {
  install.packages("BiocManager")
}

# Install Bioconductor packages only if not present
bioc_pkgs <- c("limma", "phyloseq", "microbiome", "ANCOMBC")
for (pkg in bioc_pkgs) {
  if (!requireNamespace(pkg, quietly = TRUE)) {
    BiocManager::install(pkg)
  }
}


library(qiime2R)
library(sf)
library(mapview)
library(tidyverse)
library("phyloseq")
library("ggplot2")
library("ape")
library("vegan")
library(tidyr)
library(dplyr)
library(tibble)
library("pairwiseAdonis"); packageVersion("pairwiseAdonis")
library(limma)
library(microbiome)  
library(ANCOMBC)
```

# Importing data and metadata editing
Importing and editing metadata file

```{r Importing and editing metadata file}
#import file
map <- read.csv("/Users/aiko.abo.dominguez/Desktop/Spiny_lobster_project/metadata_lobster_count24.csv")

#change the name of the first column
map$sample_id_old <- map$sample.id
colnames(map)[1] <- "sample_id"

#change "don" column to "site ID" 
colnames(map)[4] <- "site_id"

#make function to truncate ID name
truncate_before_second_underscore <- function(x) {
  sapply(strsplit(x, "_"), function(parts) paste(parts[1:2], collapse = "_"))
}

map$sample_id <- truncate_before_second_underscore(map$sample_id)
View(map)

#set column 1 as row names
rownames(map) <- map$sample_id

#make sure it is a dataframe
map <- as.data.frame(map)
View(map)

#setting variables to be numeric, as appropriate
map$Lat <- as.numeric(map$Lat)
map$Long <- as.numeric(map$Long)
map$Total_Lobsters <- as.numeric(map$Total_Lobsters)
map$EstimatedBiomass <- as.numeric(map$EstimatedBiomass)

str(map)

#add column for sample site ID based on latitude
#sites <- unique(map$Lat)
#sites
#map <- map %>%
#  mutate(site_id = ifelse(Lat==sites[1], "A",
#                   ifelse(Lat==sites[2], "B",
#                   ifelse(Lat==sites[3], "C",
#                   ifelse(Lat==sites[4], "D",
#                   ifelse(Lat==sites[5], "E",
#                   ifelse(Lat==sites[6], "F", 
#                   ifelse(Lat==sites[7],"G",
#                   ifelse(Lat==sites[8], "H",
#                   ifelse(Lat==sites[9], "I",
#                   ifelse(Lat==sites[10], "J", 
#                   ifelse(Lat==sites[11], "K",
#                   ifelse(Lat==sites[12], "L", NA)))))))))))))
unique(map$site_id)
View(map)
```
```{r Adding average mass of one lobster}
map$avg_lobster_mass <- map$EstimatedBiomass/map$Total_Lobsters
mean(map$avg_lobster_mass)
sd(map$avg_lobster_mass)
```


Exporting coordinates
```{r exporting coordinates}

coordinates <- data.frame(
  site_id = unique(map$site_id),
  lat = unique(map$Lat),
  long = unique(map$Long)
)

write.csv(coordinates,file="PAV1_eDNA_sample_site_coordinates.csv")
```



Examining metadata
```{r Examining metadata}
#plotting sample long and lat visually
mapview(map, xcol = "Long", ycol = "Lat", crs = 4269, grid = FALSE)

```
Adding habitat type to metadata from Marine Resources Geographic Information System https://myfwc.maps.arcgis.com/apps/webappviewer/index.html?id=e12fd1c0587d41c1a63eca96ec72f61a#benthic



Importing ASV sequences, taxa table, and phylogenetic tree
```{r Importing ASV sequences, taxa table, and phylogenetic tree}
#ASV table from dada2 of forward reads, only eukaryotes
ASV_qza <- read_qza("/Users/aiko.abo.dominguez/Desktop/Spiny_lobster_project/table_ASVs_lobC16S_euks_fwd.qza")
ASV_table <- as.matrix(ASV_qza$data)
#View(ASV_table)

#truncating ASV column names to match "map" truncated sample names
colnames(ASV_table) <- truncate_before_second_underscore(colnames(ASV_table))
colnames(ASV_table)
map[,"sample_id"]

#check that it worked
all(rownames(map)%in%colnames(ASV_table))

#import taxa from dada2 of forward reads
taxa_qza <- read_qza("/Users/aiko.abo.dominguez/Desktop/Spiny_lobster_project/taxonomy_fwd.qza")
taxa_table <- taxa_qza$data 
taxa_table <- taxa_table %>%
  as_tibble() %>%
  separate(Taxon, sep=";", c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")) %>%
  arrange(Feature.ID) %>%
  mutate(ASVs = paste('ASV',1:n(), sep="_")) %>%
  column_to_rownames("Feature.ID") %>%
  as.matrix()
  
#View(taxa_table)

#Importing phylogenetic tree
tree_file_qza <-read_qza("/Users/aiko.abo.dominguez/Desktop/Spiny_lobster_project/rooted_masked_aligned_seq_ASVs_lobC16S_euks_fwd.qza")
tree_file<- tree_file_qza$data
#View(tree_file)

```

# Phyloseq generation and initial filtering
Generating phyloseq
```{r Generating phyloseq}
#generating object
ps <- phyloseq(otu_table(t(ASV_table), taxa_are_rows=FALSE),
              sample_data(map), 
              tax_table(taxa_table), 
              tree_file)
ps
```

Identifying for NAs in ps filtered by taxonomic rank and turned into a df
```{r Identify NAs}
#ps_na = transform_sample_counts(ps_filter, function(x) x / sum(x)) %>%
#tax_glom("Order")%>% psmelt()

#order_NA <- subset(ps_na, Order=="NA")
#family_NA <- subset(ps_na, Family=="NA")
#genus_NA <- subset(ps_na, Genus=="NA")
#dim(order_NA)
#dim(family_NA)
#dim(genus_NA)

#View(family_NA)

#OTUs that are NA in order, but are not in genus
#setdiff(order_NA$OTU,genus_NA$OTU)

#OTUs that are NA genus, but not order
#setdiff(genus_NA$OTU,order_NA$OTU)

#this OTU is not in Order, but is in Genus
sum(genus_NA$OTU=="a9d4ae9f0286fe9bdb3881218ea6b4c1")
```


Filter "NAs"
```{r Remove 'NA' from ps}
#adding back Order to taxa missing it, had to look up by hand
taxa <- tax_table(ps)
taxa["a9d4ae9f0286fe9bdb3881218ea6b4c1", "Order"] <- "Neotaenioglossa"
taxa["8ab5f93e9ac490a760599fb46ab9f46a", "Order"] <- "Hubrechtiiformes"

#add back manually changed Order to ps
tax_table(ps) <- taxa
```


Tried to filter taxa length by site but it didn't work because i can't recombine them into one ps since now they have different branches
```{r Filter ps taxa by site}
sites <- unique(map$site_id)

#filter by site
ps_by_site <- list()

for (i in 1:length(sites)) {
  site <-map$site_id[i]
  ps_by_site[[site]] <- subset_samples(ps, site_id==site)
}

#filter taxa in each site
ps_by_site_filtered <- list()

for (i in 1:length(sites)) {
  site <-map$site_id[i]
  ps_by_site_filtered[[site]] <- filter_taxa(ps_by_site[[site]], function(x) sum(x > 10) > (0.30*length(x)), TRUE)
}

View(ps_by_site_filtered)

#combine back to one ps DOESNT WORK
#ps_filtered <- do.call("merge_phyloseq", ps_by_site_filtered)
```

Filter all ps taxa for abundance
```{r Filter all ps taxa for abundance}
#filtering taxa. Keeps ASVs where its abundance is greater than 10 in more than 10% of samples. Adjust to see what happens. 30% looks like it cuts out a lot (8879 taxa to 540), so maybe lower this?
ps_filter <- filter_taxa(ps , function(x) sum(x > 10) > (0.10*length(x)), TRUE)
ps_filter
```

# Calculating and plotting relative abundance

Relative abundance computed using custom function, sample-wise (compositional data) by ORDER for filtered  ps I think this one and the previous one do the same thing
```{r Calculate relative abundance}
#convert ps absolute ASV count into relative abundances by classification level
ps_ra = transform_sample_counts(ps_filter, function(x) x / sum(x)) %>%
tax_glom("Order")%>% psmelt()

#View(ps_ra)
dim(ps_ra)
```


Set colors for graph
```{r Set relative abundance graph colors}
#choose which ps to visualize
plot_ps <- ps_ra


#loading colors

library(RColorBrewer)
nb.cols <- length(unique(plot_ps$Order)) #set taxonomic rank
nb.cols


#set color palette, maybe better suited to more  taxonomic ranks?
mycolors <- colorRampPalette(c(brewer.pal(9, "Set1"),
                               brewer.pal(8, "Set2"),
                               brewer.pal(12, "Set3")))(nb.cols)

```

Making relative abundance graph of orders present
```{r Plot relative abundance}
RA_sup = plot_ps %>%
subset(Abundance >0.001) %>% #filter for abundance
#mutate_if(
#                is.character, 
#                stringr::str_replace_all, 
#                pattern = "o__",
#                replacement = "")  %>%

ggplot(  
       aes(x = Sample, y=Abundance, 
           fill=Order)) + #fill is  taxonomic rank

geom_bar(stat="identity", position="fill") +
scale_fill_manual(values = mycolors) +
guides(fill = guide_legend(keywidth = 0.5, , keyheight =.60, ncol=1)) +
theme_classic() + #theme
facet_grid(.~site_id, scales="free") + #separate by lattidue ("Lat")

#text sizes
theme(legend.text =element_text(size=4)) + 
theme(legend.title =element_text(size=4)) +
theme(axis.title.x = element_text(size = 5)) +
theme(axis.text.x = element_text(size =4)) +
theme(axis.title.y = element_text(size = 5)) +
theme(axis.text.y = element_text(size =4)) +
ylab("Relative Abundance") +
xlab("Sample")

RA_sup
ggsave('spiny_lobster_edna_order_siteid.png', plot=RA_sup,
       width = 14,
       height = 7,
       path = "/Users/aiko.abo.dominguez/Desktop/Spiny_lobster_project/R_code/figures")
```
CLR transformation for filtered ps for PERMANOVA and RDA

```{r CLR transformation}
ps_clr <- microbiome::transform(ps_filter, 'clr')
ps_clr
```

#Beta diversity calculations

##Beta diversity calculations natural vs block

Compute euclidian distance matrix based on the OTU/ASV table for PERMANOVA.
```{r Create distance matrix}
dis <- vegdist(otu_table(ps_clr), method ="euclidean")
#test homogeneity of group dispersions (variances).
mod <- betadisper(dis, sample_data(ps_clr)$site_type) #changed from site_id to Lat
mod
## Permutation test for F
permutest(mod)
```

PERMANOVA 
```{r Perform adnois}
#convert meta data to dataframe
df_clr= as(sample_data(ps_clr), "data.frame")
colnames(df_clr)

#running adonis2
adonis2(otu_table(ps_clr) ~ site_type, 
       data =df_clr, permutations = 999, 
         method = "euc")
```

Perform RDA
```{r RDA}
ps.ord_clr <- ordinate(ps_clr, "RDA", "euclidean")
```

Choose RDA plot colors
```{r Select RDA plot colors}
#continuous colors
mycolors <- colorRampPalette(c("red", "blue"))

#categorical colors
mycolors <- colorRampPalette(c(brewer.pal(9, "Set1"),
                               brewer.pal(3, "Set2")))
```


Plot RDA
```{r Plot RDA}
beta = plot_ordination(ps_clr, ps.ord_clr,color="site_type") +
       scale_color_manual(values=mycolors(2)) +
       theme_classic() +
       theme(legend.text =element_text(size=4)) +
       theme(legend.title = element_text(size=4)) +
       theme(axis.title.x = element_text(size = 5)) +
       theme(axis.title.y = element_text(size = 5)) +
       theme(axis.text.x = element_text(size =4)) +
       theme(axis.text.y = element_text(size =4)) +
       labs(color='Site type') 
# theme(legend.position = "none",
 #     panel.grid.minor.x = element_blank())

beta
#ggsave('spiny_lobster_ra_RDA.png', plot=beta,
   #    path = "/Users/aiko.abo.dominguez/Desktop/Spiny_lobster_project/R_code/figures")

```
## Beta diversity calculations of all sites

Compute euclidian distance matrix based on the OTU/ASV table for PERMANOVA.

```{r Create distance matrix}
dis <- vegdist(otu_table(ps_clr), method ="euclidean")
#test homogeneity of group dispersions (variances).
mod <- betadisper(dis, sample_data(ps_clr)$Lat) #changed from site_id to Lat
mod
## Permutation test for F
permutest(mod)
```

Non-metric multidimensional scaling to compare community structure among samples based on the binary dataset using Sorenson's distance matrix. Prior to PERMANOVA

Creating sorenson's distance matrix

```{r Creating sorenson's distance matrix}


```


PERMANOVA 
```{r Perform adnois}
#convert meta data to dataframe
df_clr= as(sample_data(ps_clr), "data.frame")
colnames(df_clr)

#running adonis2
adonis2(otu_table(ps_clr) ~ site_id, 
       data =df_clr, permutations = 999, 
         method = "euc")

#subset L vs A site IDs to test if it's working
ps_L_A <- subset_samples(ps_clr, site_id %in% c("A", "L"))
df_clr_LA <- subset(df_clr, site_id %in% c("A", "L"))

test <- adonis2(otu_table(ps_L_A) ~ site_id, 
       data =df_clr_LA, permutations = 999, 
         method = "euc")
test

#pairwise adonis 
devtools::install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")
library(pairwiseAdonis)

pairwise_adonis <- pairwise.adonis2(otu_table(ps_clr) ~ Don,
                 data=df_clr, permutations=999,
                 method = "euc")

#pairwise adonis using bray curtis
ps_rel <- microbiome::transform(ps,"compositional")
df_rel= as(sample_data(ps_rel), "data.frame")

dist_bray <- vegdist(df_rel,method="bray")

pairwise_adonis_bray <- pairwise.adonis2(otu_table(dis)~ site_id,
                 data=df_rel, permutations=999,
                 method = "bray")
pairwise_adonis_bray
```


Perform RDA
```{r RDA}
ps.ord_clr <- ordinate(ps_clr, "RDA", "euclidean")
```

Choose RDA plot colors
```{r Select RDA plot colors}
#continuous colors
mycolors <- colorRampPalette(c("red", "blue"))

#categorical colors
mycolors <- colorRampPalette(c(brewer.pal(9, "Set1"),
                               brewer.pal(3, "Set2")))
```


Plot RDA
```{r Plot RDA}
beta = plot_ordination(ps_clr, ps.ord_clr,color="site_id") +
       scale_color_manual(values=mycolors(12)) +
       theme_classic() +
       theme(legend.text =element_text(size=4)) +
       theme(legend.title = element_text(size=4)) +
       theme(axis.title.x = element_text(size = 5)) +
       theme(axis.title.y = element_text(size = 5)) +
       theme(axis.text.x = element_text(size =4)) +
       theme(axis.text.y = element_text(size =4)) +
       labs(color='Site ID') 
# theme(legend.position = "none",
 #     panel.grid.minor.x = element_blank())

beta
#ggsave('spiny_lobster_ra_RDA.png', plot=beta,
   #    path = "/Users/aiko.abo.dominguez/Desktop/Spiny_lobster_project/R_code/figures")

```


##ANCOMBC2
Identify taxa that have different abundances across sample sites. 

ANCOMBC2 fits a linear model to the species relative abundances that have been log-transformed across sample sites (or the variables in the chosen formula)

Description of output values of the model:

  lfc = log fold change = beta value (regression coefficient) for the first group compared to the group w/ name in the column. 
  se = standard error of lfc
  W = W-statistic, larger values indicate stronger evidence of differential abundance
  p = p-value 
  Q = PDR-corrected p-value (using Benjamini-Hochberg or similar correction)
  diff = TRUE/FALSE taxa has differential abundance based on Q
  passed_ss = TRUE/FALSE passed sample size check: whether the taxon has the minimum prevalence/sample size criteria for inclusion
  diff_robust = more conservative flag for differential abundance, controlling fro statisticall ssignificance and robustness


###ANCOMBC2 all sites
```{r Identify taxa that have different abundances across sample sites}
out <- ancombc2(
  data = ps,                        # your phyloseq object
  tax_level = "Consensus",               # or Genus/Family/etc. if aggregated
  fix_formula = "site_id",             # your variable of interest
  p_adj_method = "fdr",            # False Discovery Rate correction
  prv_cut = 0.05,     # Remove taxa with >95% zeros across samples
  lib_cut = 1000, # Filter samples with library size < 1000
  group = "site_id",               # for post-hoc contrasts (optional)
  struc_zero = TRUE,           # Identify taxa structurally absent in a group
  neg_lb = TRUE,               # Use negative lower bound correction for inference
  alpha = 0.001,               # alpha value for significance
  global = FALSE               # Don't perform global test across all levels    
)


res = out$res
View(res)
```


extract beta and standard error values from ancombc. transform SE to SD
```{r extract beta and standard error values}
#extract values
res_sort <- res %>%
  pivot_longer(
    cols = -taxon,
    names_to = c("metric", "site_id"),
    names_pattern = "^(.*)_(site_id.*|\\(Intercept\\))$",
    values_to = "value"
  ) %>%
  mutate(
    # Rename (Intercept) to a default site_id
    site_id = ifelse(site_id == "(Intercept)", "site_id1", site_id)
  ) %>%
  pivot_wider(
    names_from = metric,
    values_from = value
  )
res_sort$taxon <- res_sort
#isolate lfcs and se of only ROBUST differences in abundance
res_sig <- data.frame(lfc = res_sort$lfc*res_sort$diff_robust, #keeps only significantly abundant OTUs
                       taxon = res_sort$taxon,
                       site_id = res_sort$site_id,
                       SD = res_sort$se,
                      diff_robust = res_sort$diff_robust,
                     check.names = FALSE) %>%
  filter(diff_robust>0)#prevents R from altering the column/row names

```

Adding significant ASV info to taxon identity
```{r Adding significant ASV info to taxon identity}
library(purrr)
library(stringr)

taxa_table_sig <- as.data.frame(taxa_table) %>%
  mutate(taxon = pmap_chr(select(., 1:8), ~ str_c(..., sep = "_")))

left_join(taxa_table_sig,res_sig,by=taxon)



```

###ANCOMBC2 natural vs block
```{r Identify taxa that have different abundances across sample sites}
out <- ancombc2(
  data = ps,                        # your phyloseq object
  tax_level = "Consensus",               # or Genus/Family/etc. if aggregated
  fix_formula = "site_type",             # your variable of interest
  p_adj_method = "fdr",            # False Discovery Rate correction
  prv_cut = 0.05,     # Remove taxa with >95% zeros across samples
  lib_cut = 1000, # Filter samples with library size < 1000
  group = "site_type",               # for post-hoc contrasts (optional)
  struc_zero = TRUE,           # Identify taxa structurally absent in a group
  neg_lb = TRUE,               # Use negative lower bound correction for inference
  alpha = 0.001,               # alpha value for significance
  global = FALSE#,               # Don't perform global test across all levels  
  
 # iter_control = list(tol=1e-7, #iteration convergence tolerance for estimation-maximation algorithm
  #                    max_iter=100, #maximum # of iterations for estimation-maximation algorithm
  #                    verbose=FALSE)
)

res_type = out$res
View(res_type)

```


extract beta and standard error values from ancombc. 
```{r extract beta and standard error values}
#extract values
res_sort <- res_type %>%
  pivot_longer(
    cols = -taxon,
    names_to = c("metric", "site_type"),
    names_pattern = "^(.*)_(site_type.*|\\(Intercept\\))$",
    values_to = "value"
  ) %>%
  mutate(
    # Rename (Intercept) to a block
    site_type = ifelse(site_type == "(Intercept)", "block", "natural")
    
  ) %>%
  pivot_wider(
    names_from = metric,
    values_from = value
  ) %>%
  mutate(taxon = sub("_[^_]*$", "", taxon))

#print number of significant and non-significant taxa
length(res_sort$diff[res_sort$diff==1])
length(res_sort$diff[res_sort$diff==0])

length(res_sort$passed_ss[res_sort$passed_ss==1])
length(res_sort$passed_ss[res_sort$passed_ss==0])

#isolate lfcs and se of only ROBUST differences in abundance
res_sig <- data.frame(lfc = res_sort$lfc*res_sort$diff_robust, #keeps only significantly abundant OTUs
                       taxon = res_sort$taxon,
                       site_type = res_sort$site_type,
                       SE = res_sort$se,
                      diff_robust = res_sort$diff_robust,
                     check.names = FALSE) %>%
  filter(diff_robust>0)#prevents R from altering the column/row names

#isolate lfcs and se of only NON-ROBUST differences in abundance
res_sig <- data.frame(lfc = res_sort$lfc*res_sort$diff, #keeps only significantly abundant OTUs
                       taxon = res_sort$taxon,
                       site_type = res_sort$site_type,
                       SD = res_sort$se,
                      diff = res_sort$diff,
                     check.names = FALSE) %>%
  filter(diff>0)#prevents R from altering the column/row names

```

Adding significant ASV info to taxon identity
```{r Adding significant ASV info to taxon identity}
library(purrr)
library(stringr)

#create a reference taxa table to assign OTU to res table NVM I DONT THINK THIS WORKS BC THERE ARE MULTIPLE OTUS PER TAXA

taxa_table_sig <- as.data.frame(taxa_table) %>%
  rownames_to_column(var = "OTU") %>%
  mutate_all(~ ifelse(is.na(.), "Unknown", .)) %>%
  mutate_all(~ ifelse(.=="NA", "Unknown", .))%>% 
  mutate(taxon = pmap_chr(select(., 2:8), ~ str_c(..., sep = "_"))) 
  


res_sig <- left_join(
  res_sig,
  select(taxa_table_sig, taxon, OTU, Order),
  by = "taxon"
)



```

Graph

```{r Ancombc graph}
ancom_fig = res_sig %>%

ggplot( 
           aes(lfc, y = Order)
              ) + 
geom_point(aes(color=site_type)) +
   #geom_bar(stat = "identity", width = 1) +
           #, position = position_(width = 0.3)) +
  theme_bw() + 
 theme(legend.position = "top",
       panel.grid.minor.x = element_blank()) +
  
geom_errorbar(aes(xmin = lfc - SD, xmax = lfc + SD),
                width = 0.2,
                position = position_dodge(0.03), color = "black", alpha=0.5) + 
#scale_fill_manual(values=c("#40B5AD", "#6F8FAF")) +
 labs(y = "Order", x = "Natural vs Artificial Site (Log fold change)", color = "Site type") + 
#theme(strip.background = element_rect(fill="white")) +
scale_color_manual(values=c("#3c7658",
  "#b08f26"
)) +
theme(axis.title.x = element_text(size = 14)) +
theme(axis.title.y = element_text(size = 14)) +
theme(axis.text.y = element_text(size=8)) +
theme(axis.text.x = element_text(size =8)) +
#theme(strip.text.y = element_blank()) +
guides(color = guide_legend(keywidth = 0.2, , keyheight =.40, nrow=1)) +
#theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
theme(legend.text =element_text(size=12)) +
theme(legend.title = element_text(size=14)) +
#facet_grid(Order~., space ="free", scales="free", switch="y") +
theme(strip.background = element_blank()) +
theme(strip.text.y.left = element_text(angle =0 )) +
#theme(strip.text.y = element_text(size = 5)) +
geom_vline(xintercept = 0,  
                color = "black", linewidth=.5)

ancom_fig
ggsave("/Users/aiko.abo.dominguez/Desktop/Spiny_lobster_project/R_code/figures/PAV1_ANCOM.png",
     width = 4, height = 4, units = "in", dpi=300)
```


# Alpha diversity calculations

##Alpha diversity calculations of all sites
Filter taxa
```{r filter taxa}
#remove any ASVs with count zero
ps_div <- filter_taxa(ps , function(x) sum(x > 0) > (0.00*length(x)), TRUE)
ps_div #this didn't remove anything?
```

rareify taxa
```{r rareify taxa}
ps_rare <- phyloseq::rarefy_even_depth(ps_div, rngseed = 123, replace = TRUE, sample.size=60000)

```


Calculate alpha diversities
```{r Calculate alpha diversities}
div <- data.frame(
  "Observed" = phyloseq::estimate_richness(ps_rare, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(ps_rare, measures = "Shannon"),
  "InvSimpson" = phyloseq::estimate_richness(ps_rare, measures = "InvSimpson"),
  "Site" = phyloseq::sample_data(ps_rare)$site_id)

div
```

Rearrange data frame with alpha diversities
```{r}
#create new data frame for alpha diversity
div_new =as.data.frame(div) %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "InvSimpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "InvSimpson")))
div_new
```



Choose alpha diversity plot colors
```{r Select RDA plot colors}
#continuous colors
mycolors <- colorRampPalette(c("red", "blue"))

#categorical colors
mycolors <- colorRampPalette(c(brewer.pal(9, "Set1"),
                               brewer.pal(3, "Set2")))
```

Plot alpha diversity 
```{r Plot alpha diversity}
alpha = div_new %>%

ggplot(aes(x = Site, y = value)) +
  geom_boxplot() +
  geom_point(aes(color = Site)) +
theme_bw() +

scale_color_manual(values=mycolors(12)) +
  labs(x = "", y = "")  +
  facet_wrap(.~ metric, scales = "free") +
theme(legend.text =element_text(size=10)) +
theme(legend.title = element_text(size=10)) +
theme(axis.text.x = element_blank()) +
theme(axis.text.y = element_text(size =6)) +
theme(strip.background = element_blank()) +
theme(strip.text.x = element_text(size = 8)) 
alpha

ggsave('all_sites_alpha_div_spiny_lobster_edna.png', plot=alpha,
       width = 14,
       height = 7,
       path = "/Users/aiko.abo.dominguez/Desktop/Spiny_lobster_project/R_code/figures")
```

##Alpha diversity natural vs block

Calculate alpha diversities
```{r Calculate alpha diversities}
div_sitetype <- data.frame(
  "Observed" = phyloseq::estimate_richness(ps_rare, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(ps_rare, measures = "Shannon"),
  "InvSimpson" = phyloseq::estimate_richness(ps_rare, measures = "InvSimpson"),
  "Site type" = phyloseq::sample_data(ps_rare)$site_type)

div_sitetype
```

Rearrange data frame with alpha diversities
```{r}
#create new data frame for alpha diversity
div_new =as.data.frame(div_sitetype) %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "InvSimpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "InvSimpson")))
```


Basic stats of block vs natural biodiversity
```{r}
mean(div_sitetype$Shannon[div_sitetype$Site.type=="block"])
mean(div_sitetype$Shannon[div_sitetype$Site.type=="natural"])
sd(div_sitetype$Shannon[div_sitetype$Site.type=="block"])
sd(div_sitetype$Shannon[div_sitetype$Site.type=="natural"])
```


Choose alpha diversity plot colors
```{r Choose alpha diversity plot colors}
#continuous colors
mycolors <- colorRampPalette(c("red", "blue"))

#categorical colors
mycolors <- colorRampPalette(c(brewer.pal(9, "Set1"),
                               brewer.pal(3, "Set2")))
```


Plot alpha diversity 
```{r Plot alpha diversity}
alpha = div_new %>%

ggplot(aes(x = Site.type, y = value)) +
  geom_boxplot() +
  geom_point(aes(color = Site.type)) +
theme_bw() +

scale_color_manual(values=mycolors(2),
                   labels=c(block="Cinder Block",natural="Natural")) +
  labs(x = "", y = "", color="Site Type")  +
  facet_wrap(.~ metric, scales = "free") +
theme(legend.text =element_text(size=10)) +
theme(legend.title = element_text(size=10)) +
theme(axis.text.x = element_blank()) +
theme(axis.text.y = element_text(size =6)) +
theme(strip.background = element_blank()) +
theme(strip.text.x = element_text(size = 8)) 
alpha

ggsave('site_type_alpha_div_spiny_lobster_edna.png', plot=alpha,
       width = 14,
       height = 7,
       path = "/Users/aiko.abo.dominguez/Desktop/Spiny_lobster_project/R_code/figures")
```